src/main/java/com/wavefront/jersey/sdk/reporter/WavefrontJerseyReporter.java

```diff
@@ -0,0 +1,165 @@
+package com.wavefront.jersey.sdk.reporter;
+
+import com.wavefront.internal.reporter.WavefrontInternalReporter;
+import com.wavefront.internal_reporter_java.io.dropwizard.metrics5.MetricName;
+import com.wavefront.jersey.sdk.ApplicationTags;
+import com.wavefront.sdk.common.NamedThreadFactory;
+import com.wavefront.sdk.common.WavefrontSender;
+import com.wavefront.sdk.entities.metrics.WavefrontMetricSender;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Logger;
+
+import javax.annotation.Nullable;
+
+import static com.wavefront.jersey.sdk.Constants.HEART_BEAT_METRIC;
+import static com.wavefront.jersey.sdk.Constants.JERSEY_SERVER_COMPONENT;
+import static com.wavefront.jersey.sdk.Constants.NULL_TAG_VAL;
+
+/**
+ * Wavefront reporter for your Jersey based application responsible for reporting metrics and
+ * histograms out of the box for you.
+ *
+ * @author Sushant Dewan (sushant@wavefront.com).
+ */
+public class WavefrontJerseyReporter implements JerseyReporter, Runnable {
+
+  private static final Logger logger =
+      Logger.getLogger(WavefrontJerseyReporter.class.getCanonicalName());
+  private final WavefrontInternalReporter wfReporter;
+  private final WavefrontMetricSender wavefrontMetricSender;
+  private final ApplicationTags applicationTags;
+  private final String source;
+
+  private WavefrontJerseyReporter(WavefrontInternalReporter wfReporter,
+                                  WavefrontMetricSender wavefrontMetricSender,
+                                  ApplicationTags applicationTags,
+                                  String source) {
+    this.wfReporter = wfReporter;
+    this.wavefrontMetricSender = wavefrontMetricSender;
+    this.applicationTags = applicationTags;
+    this.source = source;
+    ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1,
```

Comment:
Can we extract this to a common class at this point, at some point we would pull this to T1.